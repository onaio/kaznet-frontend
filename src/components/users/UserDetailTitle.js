// Renders the detail page title section
import React, { Component } from "react";
import { Row, Col, Modal, ModalHeader, ModalFooter, Button } from "reactstrap";
import { Link } from "react-router-dom";
import Moment from "react-moment";

import "../page/DetailTitle.css";

export default class UserDetailTitle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <section className="detail-page-title">
        <Col sm="12" md={{ size: 8, offset: 2 }}>
          <div>
            <h1 className="kaznet-detail-title">
              <Link to="/users" className="kaznet-header-link">
                Users
              </Link>{" "}
              > {this.props.user.attributes.ona_username}
            </h1>
            <p className="kaznet-creation-detail">
              {"By "}
              {this.props.user.attributes.created_by_name},{" "}
              <Moment format="DD-MM-YYYY">
                {this.props.user.attributes.created}
              </Moment>
            </p>
            <Col md="12">
              <Row className="kaznet-action-links">
                <Link
                  to={`/users/${this.props.user.attributes.id}/edit`}
                  className="action-link"
                >
                  EDIT
                </Link>&nbsp;&nbsp;|&nbsp;&nbsp;
                <Button
                  color="link"
                  className="remove_button_css action-link action-link-alert"
                  onClick={this.toggle}
                >
                  DELETE USER
                </Button>
                <Modal
                  isOpen={this.state.modal}
                  toggle={this.toggle}
                  className={this.props.className}
                >
                  <ModalHeader toggle={this.toggle}>
                    Are you sure you want to delete this USER?
                  </ModalHeader>
                  <ModalFooter>
                    <Link
                      to={`/users/${this.props.user.id}/delete`}
                      className="btn btn-danger"
                      onClick={this.toggle}
                    >
                      Delete User
                    </Link>
                    <Button color="secondary" onClick={this.toggle}>
                      Cancel
                    </Button>
                  </ModalFooter>
                </Modal>
              </Row>
            </Col>
          </div>
        </Col>
      </section>
    );
  }
}
