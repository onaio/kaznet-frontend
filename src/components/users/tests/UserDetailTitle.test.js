// Test UserDetailTitle
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Router } from "react-router";
import createBrowserHistory from "history/createBrowserHistory";

import ErrorBoundary from "../../ErrorBoundary";
import { singleUserData } from "../../../store/users/tests/fixtures";
import UserDetailTitle from "../UserDetailTitle";

const history = createBrowserHistory();

describe("components/users/UserDetailTitle", () => {
  it("renders without crashing", () => {
    shallow(
      <ErrorBoundary>
        <UserDetailTitle user={singleUserData} />
      </ErrorBoundary>
    );
  });
  it("renders the correct html", () => {
    const wrapper = mount(
      <ErrorBoundary>
        <Router history={history}>
          <UserDetailTitle user={singleUserData} />
        </Router>
      </ErrorBoundary>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.unmount();
  });
});
