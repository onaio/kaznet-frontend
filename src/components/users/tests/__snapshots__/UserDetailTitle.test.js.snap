// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/users/UserDetailTitle renders the correct html 1`] = `
<ErrorBoundary>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UserDetailTitle
      user={
        Object {
          "attributes": Object {
            "amount_earned": "0 KES",
            "approval_rate": 0,
            "approved_submissions": 0,
            "avg_amount_earned": 0,
            "avg_approval_rate": 0,
            "avg_approved_submissions": 0,
            "avg_rejected_submissions": 0,
            "avg_submissions": 0,
            "created": "2018-06-19T16:03:10.184043+03:00",
            "email": "sol@admin.me",
            "expertise": "1",
            "first_name": "Davis",
            "gender": "1",
            "last_login": "2018-06-19T16:03:22+03:00",
            "last_name": "Raymond",
            "modified": "2018-06-20T08:45:12.976916+03:00",
            "national_id": null,
            "ona_pk": null,
            "ona_username": "davisraym",
            "payment_number": "",
            "phone_number": "",
            "rejected_submissions": 0,
            "role": "1",
            "role_display": "Admin",
            "submission_count": 0,
          },
          "id": "1",
          "type": "UserProfile",
        }
      }
    >
      <section
        className="detail-page-title"
      >
        <Col
          md={
            Object {
              "offset": 2,
              "size": 8,
            }
          }
          sm="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <div
            className="col-sm-12 col-md-8 offset-md-2"
          >
            <div>
              <h1
                className="kaznet-detail-title"
              >
                <Link
                  className="kaznet-header-link"
                  replace={false}
                  to="/users"
                >
                  <a
                    className="kaznet-header-link"
                    href="/users"
                    onClick={[Function]}
                  >
                    Users
                  </a>
                </Link>
                 
                &gt; 
                davisraym
              </h1>
              <p
                className="kaznet-creation-detail"
              >
                By 
                ,
                 
                <t
                  ago={false}
                  calendar={false}
                  decimal={false}
                  element="time"
                  filter={[Function]}
                  format="DD-MM-YYYY"
                  fromNow={false}
                  interval={60000}
                  onChange={[Function]}
                  toNow={false}
                  unit={null}
                  unix={false}
                  utc={false}
                >
                  <time
                    dateTime={"2018-06-19T13:03:10.184Z"}
                  >
                    19-06-2018
                  </time>
                </t>
              </p>
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-12"
                >
                  <Row
                    className="kaznet-action-links"
                    tag="div"
                  >
                    <div
                      className="kaznet-action-links row"
                    >
                      <Link
                        className="action-link"
                        replace={false}
                        to="/users/edit/undefined"
                      >
                        <a
                          className="action-link"
                          href="/users/edit/undefined"
                          onClick={[Function]}
                        >
                          EDIT
                        </a>
                      </Link>
                        |  
                      <Modal
                        autoFocus={true}
                        backdrop={true}
                        backdropTransition={
                          Object {
                            "mountOnEnter": true,
                            "timeout": 150,
                          }
                        }
                        centered={false}
                        fade={true}
                        isOpen={false}
                        keyboard={true}
                        modalTransition={
                          Object {
                            "timeout": 300,
                          }
                        }
                        onClosed={[Function]}
                        onOpened={[Function]}
                        role="dialog"
                        toggle={[Function]}
                        zIndex={1050}
                      />
                    </div>
                  </Row>
                </div>
              </Col>
            </div>
          </div>
        </Col>
      </section>
    </UserDetailTitle>
  </Router>
</ErrorBoundary>
`;
